RandLAPACK/misc/rl_linops.hh

```diff
@@ -15,6 +16,130 @@
 
 namespace RandLAPACK::linops {
 
+// Abstract linear operator concept
+template<typename LinOp, typename T = LinOp::scalar_t>
+concept LinearOperator = requires(LinOp A) {
+    { A.n_rows }  -> std::same_as<const int64_t&>;
+    { A.n_cols }  -> std::same_as<const int64_t&>;
+} && requires(LinOp A, Layout layout, Op trans_A, Op trans_B, int64_t m, int64_t n, int64_t k, T alpha, int64_t lda, T* const B, int64_t ldb, T beta, T* C, int64_t ldc) {
+    // A Matmul-like function that updates C := alpha A*B + beta C, where
+    // B and C have n columns and are stored in layout order with strides (ldb, ldc).
+    { A(layout, trans_A, trans_B, m, n, k, alpha, lda, B, ldb, beta, C, ldc) } -> std::same_as<void>;
+};
+
+// Sparse linear operator struct, supplied with a sparse-with-dense matrix multiplication operator 
+// (with sparse input on the left) and a Frobenius norm function.
+template <typename T, RandBLAS::sparse_data::SparseMatrix SpMat>
+struct SpLinOp {
+    using scalar_t = T;
+    const int64_t n_rows;
+    const int64_t n_cols;
+    SpMat &A_sp_buff;
```

Comment:
Naming convention: don't call it ``A_sp_buff``, since it isn't a buffer.

Reply:
Fixed.